Dependencies/CMakeLists.txt

```diff
@@ -0,0 +1,131 @@
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
I'm not sure I understand the point of this file. Are we expecting that anyone who wants to use any dependencies add this as a subdirectory in CMakeLists.txt?

Reply:
Spoke offline. This file allows the Dependencies folder to participate in the CMake pattern shared by the other major folders (Core, Plugins, and Apps). Most of the code in here is currently just ported directly from the old build system, but hopefully we'll improve it as we go to make it as clean and airtight as possible over time.